Adding support to HeartBeat Monitor type

rogerioefonseca opened the pull request:
- [x] Add support to heartbeat
- [x] Update documentation
- [x] Refactor to reduce number of lines
- [x] test the new monitor type

rogerioefonseca commented:
@mnaser PTAL on this new MR.
I was working in a fork from louy repository and that one was some commits behind yours, now, I forked yours and created a new MR based on this.
And then the conflicts were gone

mnaser commented on docs/resources/monitor.md:
```diff
@@ -16,44 +16,45 @@ resource "uptimerobot_monitor" "my_website" {
       - `imap`
       - `custom`
-    - `port` - the port monitored (only if subtype is `custom`)
-* `http_method` - available for HTTP monitoring. Can be one of the following:
-  - `HEAD` (default for non-keyword)
-  - `GET` (default for keyword)
-  - `POST`
-  - `PUT`
-  - `PATCH`
-  - `DELETE`
-  - `OPTIONS`
```
I think this was accidentally swallowed in the refactor :)

rogerioefonseca replied:
Yep, you are right, thanks for the review.
Fixed  - 84523438ede1594ed18f7fbc921770315f5ea895

mnaser reviewed (commented):
one tiny nit but the refactor looks great.

mnaser commented:
It looks like the tests failed, could you perhaps try and run them locally and see if you can reproduce and fix this?

I appreciate your follow up on this!

rogerioefonseca commented:
> It looks like the tests failed, could you perhaps try and run them locally and see if you can reproduce and fix this?
> 
> 
> 
> I appreciate your follow up on this!

I'll do it later today.

rogerioefonseca commented:
> > It looks like the tests failed, could you perhaps try and run them locally and see if you can reproduce and fix this?
> > I appreciate your follow up on this!
> 
> I'll do it later today.

Done

rogerioefonseca commented:
> > It looks like the tests failed, could you perhaps try and run them locally and see if you can reproduce and fix this?
> > I appreciate your follow up on this!
> 
> I'll do it later today.

It was super late, and I misread the test outputs, now that I saw your commit I realized the tests problems.
But nice that you figure those and applied to master. I rebased and added your commits to this MR

mnaser commented:
> BTW, is there a special reason for the tests do not run right after the pull request is created?
> I saw in the workflow that the tests are limited to be executed based on a cron definition...

It's a GitHub thing since you have not had any contributions yet.

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

rogerioefonseca commented:
> > BTW, is there a special reason for the tests do not run right after the pull request is created?
> > I saw in the workflow that the tests are limited to be executed based on a cron definition...
> 
> It's a GitHub thing since you have not had any contributions yet.
> 
> https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

Looks like it needs to add the TF and API KEY env variables into the Repo configuration, for some reason that I do not know looks like those are not being loaded into this one.

Will you have time to TAL on this.
Let me know if I can help somehow.

FlxPeters commented:
Any progress here? Would like to use this feature too and this seams to be the best maintained fork of the original repo. 
Let me know if i can help bring this to an successful merge.

rogerioefonseca commented:
> Any progress here? Would like to use this feature too and this seams to be the best maintained fork of the original repo. 
> 
> Let me know if i can help bring this to an successful merge. 

@mnaser ....

rogerioefonseca commented:
> Any progress here? Would like to use this feature too and this seams to be the best maintained fork of the original repo. 
> 
> Let me know if i can help bring this to an successful merge. 

 Not sure what else can I do as well.
@mnaser, I'm willing to help to get this merged as well.

rogerioefonseca commented:
> > Any progress here? Would like to use this feature too and this seams to be the best maintained fork of the original repo. 
> > 
> > Let me know if i can help bring this to an successful merge. 
> 
>  Not sure what else can I do as well.
> @mnaser, I'm willing to help to get this merged as well.

Ping...

dcardellino commented:
Any Update here? Would like to use this feature to not do this manually...
